### ISSUE TITLE
Additional Announcement for the midterm exam.

### ISSUE THREAD

**Issue by Lee-Janggun** [kaist-cp/cs220#407]
This is an additional announcement for the midterm. Please read it carefully, and make sure to ask any questions you have.

### Charging Laptop

* **Bring your laptop in a fully charged state**.
* We cannot guarantee that you will be able to charge your laptop during the exam, so please make sure to bring your laptop in a fully charged state.
* If your battery runs out during the midterm exam, we cannot help you.

### Exam Time & Structure

The exam will be from 1:00PM to 3:10PM, and will consist of three parts

* Part 1: T/F questions (30 minutes)
* Intermission: chaning SEB from part 1 to part 2 (10 minutes)
* Part 2: Short answer questions (90 minutes)

Once we start intermission, **you won't be able to go back to part 1**.

Each part may start a bit late depending on how to exam goes (e.g., start on 1:05 PM for part 1, and 2:45 PM for part 2).

### Network
~~We understand that the network at 1501 is not that stable. We will prepare additional routers for better network connection during the exam.~~

Update: the CS administration team has enhanced the E3_1501 network. We will use that for the exam.

### Short Answer Questions.

TL;DR

* **You must only submit the code for `todo!()`**
* **If `todo!()` replaced with your submission doesn't compile, you get 0 points.**
* **There will be hidden test cases**

* The short answer question will be of the following form:
<img width="648" alt="short-answer" src="https://github.com/user-attachments/assets/49d6b168-4b3e-4624-8b98-8b7d8716fc1a">

* In this question, your submitted answer to the Google Form should exactly the replacement for `todo!()` in the code snippet.
* For example, a *correct* submission only enters the code `input`:
<img width="653" alt="short-answer-correct" src="https://github.com/user-attachments/assets/c473edd7-9441-4b34-aba0-27f1a6ed5fb0">
  * You don't need to care about indent or format. For example, the following submission with a lot of spaces in front will work.
  
                                              input

* Any submission that submits more codes will be marked incorrect.
* For example, an *incorrect* submission submits the entire function `id_usize`.
<img width="642" alt="short-answer-incorrect" src="https://github.com/user-attachments/assets/34072f20-c818-4d47-bebf-7777f3852308">

* If there are more than one `todo!()`, submit the answer for each `todo!()` separately.
<img width="647" alt="short-answer-multiple" src="https://github.com/user-attachments/assets/9d2eba73-5ac5-41ce-bb6e-a155033ef34a">

**Comment by mhchung04** [comment]
If the answer required to give several lines, how should I indent?
When I should answer the code including, 
println!("Hello");
println!("World");

then should I submit
println!("Hello");
println!("World");

or whether
println!("Hello");
____println!("World");

**Comment by Lee-Janggun** [comment]
@mhchung04 Do the first, the second won't compile.

**Comment by mookyung-choi** [comment]
In google form, the answer should be as follows
```
    println!("hello, world!");
```

I am confused about the submission format.
For example, In the first question of Rehearsal part 2, What format should I submit it in google form?

''''
    let mut sum = 0;
    for i in input {
        sum += i;
    }
    sum
''''

or

let mut sum = 0;
for i in input {
    sum += i;
}
sum

**Comment by Lee-Janggun** [comment]
@mookyung-choi. They will both work, you don't need to care about indent or format (I'll update the issue to explicitly mention this).

**Comment by ICubE-** [comment]
For questions asking about runtime behaviors, is the runtime environment ideal?
In other words, is the memory infinite?

Example:
This is a question from previous exam.
I think this code terminates for all `a`, but it aborts with stack overflow for some inputs in my PC.
```rust
// Question: `foo` function terminates for all `a`.
fn foo(a: i32) -> i32 {
    if a < 10 {
        return a;
    }

    foo(a.wrapping_add(1))
}
```

**Comment by Lee-Janggun** [comment]
You can assume infinite heap space/stack space **unless specified otherwise**.
